### DISCUSSION TITLE
Allow using alias directives to reference other using aliases

### DISCUSSION THREAD

**Discussion by TheUnlocked** [dotnet/csharplang#2896] [Language Ideas]
Aliases should be able to reference other aliases. For example, this is not legal:
```cs
using Chunk = Chunk<bool, byte>;
using ChunkMap = Dictionary<Point, Chunk>;
```
But this is:
```cs
using Chunk = Chunk<bool, byte>;
using ChunkMap = Dictionary<Point, Chunk<bool, byte>>;
```
Ideally if you want to change something about the definition of `Chunk` (e.g. changing `byte` to `short`), you should be able to do it in one place rather than needing to do it for all of the aliases which use the same definition.

To show that this isn't some weird thing with generics and using the same name as the generic type, this also doesn't work:
```cs
using F = Foo;
using B = F.Bar;

public class Foo{
    public class Bar{

    }
}
```
Changing `using B = F.Bar;` to `using B = Foo.Bar;` compiles fine.

Edit:
This could be a breaking change in some cases, but I still think this has benefits which should be considered, even if not exactly using the proposed syntax. Or perhaps types/namespaces with the same name as an alias should just get precedence over the alias to avoid any breaking changes. There are multiple ways to get around this issue.

**Comment by huoyaoyuan** [comment]
This can helps to simplify using alias of long generics.
```csharp
using SomeClass = Some.VeryLong.Namespace.GenericType
    <Some.VeryLong.Namesmace.SomeVeryLongClass,
    Some.VeryLong.Namesmace.SomeVeryLongInterface>
```

**Comment by LouisFr81** [comment]
You can reference other aliases, if you declare the second inside a namespace:
```C#
using Chunk = Chunk<bool, byte>;
namespace MyNamespace
{
    using ChunkMap = Dictionary<Point, Chunk>;
